src/Pages/Register.cshtml.cs

```diff
@@ -36,6 +37,12 @@ public RegisterModel(UserDatabase userDatabase, EmailSender emailSender)
         {
             UserDatabase = userDatabase;
             EmailSender = emailSender;
+            UserDatabase.Registered += OnRegister;
+        }
+
+        private async void OnRegister(object sender, UserModel user)
+        {
+            await EmailSender.SendGreetingEmailAsync(user.Email);
         }
```

Comment:
Seems neat to send the greetings email in the first place, second nice that events were used :D